app/scripts/inpage.js

```diff
@@ -1,3 +1,9 @@
+/* eslint-disable import/first */
+import log from 'loglevel';
+import { WindowPostMessageStream } from '@metamask/post-message-stream';
+import { initializeProvider } from '@metamask/providers/dist/initializeInpageProvider';
+import shouldInjectProvider from '../../shared/modules/provider-injection';
+
 // need to make sure we aren't affected by overlapping namespaces
```

Comment:
Why were these imports moved? They were intentionally delayed to resolve an incompatibility, which is what this comment is about. I don't remember the details though.

Reply:
This change was an artifact of me trying to conditionally import and execute all provider code, because `import` needed to be top level.  I've put the imports back.